pkg/listwatch/listinformer.go

```diff
@@ -130,6 +130,10 @@ func (li *listerInformer) stop() {
 }
 
 func (li *listerInformer) newWatchEvent(eventType watch.EventType, obj interface{}) *watch.Event {
+	tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+	if ok {
+		obj = tombstone.Obj
+	}
 	return &watch.Event{
```

Comment:
Worth to add another check like 
```go
if _, ok := obj.(runtime.Object); !ok {
   print err; return nil
}
```
before trying to use it below?

Reply:
Sounds like a good idea

Reply:
@jwendell should be handled now